src/bot/storage/sheets.ts

```diff
@@ -22,6 +22,54 @@ const {
 
 const worksheetName = WORKSHEET_NAME || "_moneyman";
 
+/**
+ * Retry function for Google API operations with exponential backoff
+ * Handles transient errors like 503 "Service is currently unavailable"
+ */
+async function retryGoogleApiOperation<T>(
```

Comment:
1. Since we already depend on the `async` package, just use https://caolan.github.io/async/v3/docs.html#retry
2. I trust Google to have 503 in the error, and (if available on the axius errors, to have 503 as the status,) so you can have less error checks